modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/utils/openCustomDateModal.tsx

```diff
@@ -0,0 +1,170 @@
+import classNames from 'classnames';
+import {useId} from 'frontend-js-components-web';
+import {navigate} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+const NOT_SELECTED_OPTION = {
+	label: `-- ${Liferay.Language.get('select-type')} --`,
+	value: 'not-selected',
+};
+
+function getDateRange(start: string | undefined, end: string | undefined) {
+	if (start && end) {
+		return `${start} - ${end}`;
+	}
+
+	return undefined;
+}
+
+function isValidDate(date: string) {
+	return !isNaN(new Date(date).getDate());
+}
+
+type Props = {
+	dateTypes: Array<{label: string; value: string}>;
+	filterUrl: string;
+	namespace: string;
+	selectedDateType: string | undefined;
+	selectedEndDate: string | undefined;
+	selectedStartDate: string | undefined;
+};
+
+export default function openCustomDateModal(props: Props) {
+	render(CustomDateModal, {...props}, document.createElement('div'));
+}
+
+function CustomDateModal({
+	dateTypes,
+	filterUrl,
+	namespace,
+	selectedDateType,
+	selectedEndDate,
+	selectedStartDate,
+}: Props) {
+	const {observer, onOpenChange, open} = useModal({
+		defaultOpen: true,
+		onClose: () => onOpenChange(false),
+	});
+
+	const dateTypeId = useId();
+	const rangeId = useId();
+
+	const [dateType, setDateType] = useState(selectedDateType);
+	const [endDate, setEndDate] = useState(selectedEndDate);
+	const [startDate, setStartDate] = useState(selectedStartDate);
+
+	const [hasError, setHasError] = useState(false);
+
+	const onSave = () => {
+		if (!isValidDate(startDate!) || !isValidDate(endDate!)) {
+			setHasError(true);
+
+			return;
+		}
+
+		const url = new URL(filterUrl);
+
+		url.searchParams.set(`${namespace}dateType`, dateType!);
+		url.searchParams.set(`${namespace}startDate`, startDate!);
+		url.searchParams.set(`${namespace}endDate`, endDate!);
+
+		navigate(url.toString());
+	};
+
+	if (!open) {
+		return null;
+	}
+
+	return (
+		<ClayModal observer={observer}>
+			<ClayModal.Header>
+				{Liferay.Language.get('filter-by-date')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				<ClayForm.Group>
+					<label htmlFor={dateTypeId}>
+						{Liferay.Language.get('date-type')}
+					</label>
+
+					<ClaySelectWithOption
+						id={dateTypeId}
+						onChange={(event) => setDateType(event.target.value)}
+						options={[NOT_SELECTED_OPTION, ...dateTypes]}
+						value={dateType || NOT_SELECTED_OPTION.value}
+					/>
+				</ClayForm.Group>
+
+				{dateType ? (
+					<ClayForm.Group
+						className={classNames({'has-error': hasError})}
+					>
+						<label htmlFor={rangeId}>
+							{Liferay.Language.get('date-range')}
+						</label>
+
+						<ClayDatePicker
+							onChange={(range: string) => {
+								const [start, end] = range!.split(' - ');
+
+								setStartDate(start);
+								setEndDate(end);
+
+								setHasError(false);
+							}}
+							placeholder="YYYY-MM-DD - YYYY-MM-DD"
+							range
+							value={getDateRange(startDate, endDate)}
+							years={{
+								end: 2024,
```

Comment:
I guess that this shouldn't be hardcoded. `end: new Date().getFullYear()` would be the best, and maybe `new Date().getFullYear() - 10` ? for the start date?

Reply:
Totally agree, copied it from the Clay example and forgot to change it 😅 will adapt it, thanks Ambrin!